lib/node_modules/@stdlib/_tools/changelog/parse-commits/lib/conventional_changelog.js

```diff
@@ -0,0 +1,276 @@
+	*
+	* @private
+	* @param {Object} _ - AST node
+	* @param {ObjectArray} ancestors - node ancestors
+	*/
+	function processBreakingChanges( _, ancestors ) {
+		var startCollecting = false;
+		var breaking = {
+			'title': 'BREAKING CHANGE',
+			'text': ''
+		};
+		var parent = ancestors.pop();
+		switch ( parent.type ) {
+		case 'summary':
+			breaking.text = out.subject;
+			break;
+		case 'body':
+			breaking.text = '';
+			visit( parent, [ 'text', 'breaking-change' ], visitBodyNode );
+			break;
+		case 'token':
+			parent = ancestors.pop();
+			visit( parent, 'text', visitTextNode );
+			break;
+		default:
+			break;
+		}
+		if ( breaking.text !== '' ) {
+			out.notes.push( breaking );
+		}
+
+		/**
+		* Callback invoked upon visiting a body node.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function visitBodyNode( node ) {
+			if ( startCollecting && node.type === 'text' ) {
+				if ( breaking.text !== '' ) {
+					breaking.text += '\n';
+				}
+				breaking.text += node.value;
+			} else if ( node.type === 'breaking-change' ) {
+				startCollecting = true;
+			}
+		}
+
+		/**
+		* Callback invoked upon visiting a text node.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function visitTextNode( node ) {
+			breaking.text = node.value;
+		}
+	}
+
+	/**
+	* Callback invoked upon visiting the body node.
+	*
+	* @private
+	* @param {Object} body - AST node
+	*/
+	function onBody( body ) {
+		if ( body ) {
+			visit( body, 'text', extractBody );
+		}
+
+		/**
+		* Callback invoked upon visiting a text node.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function extractBody( node ) {
+			if ( out.body !== '' ) {
+				out.body += '\n';
+			}
+			out.body += node.value;
+		}
+	}
+
+	/**
+	* Callback invoked upon visiting the summary node.
+	*
+	* @private
+	* @param {Object} summary - AST node
+	*/
+	function onSummary( summary ) {
+		visit( summary, visitSummary );
+
+		/**
+		* Processes a summary node and extracts relevant information.
+		*
+		* @private
+		* @param {Object} node - AST node
+		*/
+		function visitSummary( node ) {
+			switch ( node.type ) {
+			case 'type':
+				out.type = node.value;
+				out.header += node.value;
+				break;
+			case 'scope':
+				out.scope = node.value;
+				out.header += '(' + node.value + ')';
+				break;
+			case 'breaking-change':
+				out.header += '!';
+				break;
+			case 'text':
+				out.subject = node.value;
+				out.header += ': ' + node.value;
+				break;
+			default:
+				break;
+			}
+		}
+	}
+
+	/**
+	* Processes a footer node and extracts relevant information.
+	*
+	* @private
+	* @param {Object} node - AST node
+	*/
+	function processFooter( node ) {
+		var hasRefSepartor = false;
+		var coAuthoredBy = false;
+		var closesIssue = false;
+		var reference = {};
+		var hasPRURL = false;
+		visit( node, [ 'type', 'separator', 'text' ], processFooterNode );
+		if ( hasRefSepartor && reference.ref.match( RE_DIGITS ) ) {
```

Comment:
Why use `match` here? Shouldn't you use `RE_DIGITS.test()`? You're not doing anything with the `match` return value which is wasteful from a micro-perf POV.

Reply:
Changed!